Doc/extending/newtypes_tutorial.rst

```diff
@@ -0,0 +1,865 @@
+.. highlightlang:: c
+
+.. _defining-new-types:
+
+**********************************
+Defining Extension Types: Tutorial
+**********************************
+
+.. sectionauthor:: Michael Hudson <mwh@python.net>
+.. sectionauthor:: Dave Kuhlman <dkuhlman@rexx.com>
+.. sectionauthor:: Jim Fulton <jim@zope.com>
+
+
+Python allows the writer of a C extension module to define new types that
+can be manipulated from Python code, much like the built-in :class:`str`
+and :class:`list` types.  This is not hard; the code for all extension types
```

Comment:
I don't think we need the "This is not hard" comment here, as for folks that aren't experienced C programmers, it *is* going to be hard.

We may also want to repeat the cross-references to Cython/CFFI/etc, and perhaps point to https://packaging.python.org/guides/packaging-binary-extensions/#an-overview-of-binary-extensions for further discussion of the available alternatives.

Reply:
Agreed. I'll remove that wording.